README.md

```diff
@@ -416,6 +417,64 @@ cargo run -- --state-archive-capacity <CAPACITY>
 
 All RPC endpoints that support querying the state at an old (non-latest) block only work with state archive capacity set to `full`.
 
+## Account impersonation
+
+Devnet allows you to use impersonated account from mainnet/testnet. To use the feature it is required to start devnet in forked mode.
+Account impersonation supports 4 methods:
+ - Impersonate specific account address, non existing in the local state  (devnet_impersonateAccount)
+ - Automatic impersonation of an account, non existing in the local state (devnet_stopImpersonateAccount)
+ - Stop impersonation of an account (devnet_autoImpersonate)
+ - Stop auto impersonation (devnet_stopAutoImpersonate)
```

Comment:
[a7c2872](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/436/commits/a7c2872d6f32eac4a03bae737a00fcbdf6f95f31)

Reply:
The documentation is now much clearer, but I'm wondering: Is `invalid signature` really the error that should be thrown in case impersonation is not used?

Reply:
what you have in mind

Reply:
What is the full error? Is it just `invalid signature`? I mean: if a user attempts to send a transaction with a non-deployed account, what is:
1. the best error to be returned UX-wise?
2. the error returned by testnet/mainnet?

Reply:
"Execution failed. Failure reason: 0x4163636f756e743a20696e76616c6964207369676e6174757265 ('Account: invalid signature')."

Reply:
if a user attempts to send a transaction with a non-deployed account, what is:
> 
> 1. the best error to be returned UX-wise?
> 2. the error returned by testnet/mainnet? not enough balance

Reply:
what do you mean

Reply:
I mean, should we have the same response returned by Devnet? Or are you happy with what we have now?

Reply:
When a user sends a declare cairo 0 transaction with non-deployed account then the error returned is UninitializedStorageAddress ( to rpc: https://starknet-sepolia.public.blastapi.io/rpc/v0_7) . But on devnet sending a declare and invoke transaction returns InsufficientAccountBalance.

Reply:
Conclusion: we leave the functionality as it is, but update the docs

Reply:
[66f3273](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/436/commits/66f3273a67c91beddfa990ccb6451b16029db930)